src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -38,6 +31,48 @@ public static IServiceCollection AddSignalRServiceManager(this IServiceCollectio
             return services.AddSignalRServiceManager();
         }
 
+        /// <summary>
+        /// Adds the essential SignalR Service Manager services to the specified services collection.
+        /// </summary>
+        /// <remarks>Designed for Azure Function extension where the setup of <see cref="ServiceManagerOptions"/> is different from SDK</remarks>
+        /// <typeparam name="TOptionsSetup">The type of class used to setup <see cref="ServiceManagerOptions"/>. </typeparam>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static IServiceCollection AddSignalRServiceManager<TOptionsSetup>(this IServiceCollection services) where TOptionsSetup : class, IConfigureOptions<ServiceManagerOptions>, IOptionsChangeTokenSource<ServiceManagerOptions>
+        {
+            //cascade options setup
+            services.AddSingleton<TOptionsSetup>()
+                    .AddSingleton<IConfigureOptions<ServiceManagerOptions>>(sp => sp.GetService<TOptionsSetup>())
+                    .AddSingleton<IOptionsChangeTokenSource<ServiceManagerOptions>>(sp => sp.GetService<TOptionsSetup>());
+            services.PostConfigure<ServiceManagerOptions>(o => o.ValidateOptions());
+            services.AddSingleton<ServiceManagerContextSetup>()
+                    .AddSingleton<IConfigureOptions<ServiceManagerContext>>(sp => sp.GetService<ServiceManagerContextSetup>())
+                    .AddSingleton<IOptionsChangeTokenSource<ServiceManagerContext>>(sp => sp.GetService<ServiceManagerContextSetup>());
+
+            services.AddSignalR()
+                    .AddAzureSignalR<ServiceOptionsSetup>();
+
+            //add dependencies for persistent mode only
+            services
+                .AddSingleton<ConnectionFactory>()
+                .AddSingleton<IConnectionFactory>(sp =>
+                {
+                    var productInfo = sp.GetRequiredService<IOptions<ServiceManagerContext>>().Value.ProductInfo;
+                    var defaultConnectionFactory = sp.GetRequiredService<ConnectionFactory>();
+                    return new ManagementConnectionFactory(productInfo, defaultConnectionFactory);
```

Comment:
can it change to `ManagementConnectionFactory(ServiceManagerContext c, ConnectionFactory f)`? 
and then `.AddSingleton<IConnectionFactory, ManagementConnectionFactory>`?

Reply:
Yes